src/arizona_template_compiler.erl

```diff
@@ -125,22 +125,23 @@ compile_expr(ExprStr, Loc, T, State) ->
 
 expand_expr(Expr, Loc, Eval, State) ->
     ExprTree = merl:quote(Expr),
+    Macros = State#state.macros,
+    MacrosEnv = [{K, merl:term(V)} || K := V <- Macros],
     AttrsEnv = lists:map(fun
         ({K, {expr, _KLoc, KExpr}}) ->
-            {binary_to_existing_atom(K, utf8), merl:quote(KExpr)};
+            KForm = merl:tsubst(merl:quote(KExpr), MacrosEnv),
+            {binary_to_existing_atom(K, utf8), KForm};
         ({K, {text, _KLoc, KTxt}}) ->
-            {binary_to_existing_atom(K, utf8),
+            {binary_to_atom(K, utf8),
               merl:quote(iolist_to_binary(["<<\"", KTxt, "\"/utf8>>"]))}
     end, State#state.attributes),
     AttrsTree = merl:tsubst(ExprTree, AttrsEnv),
-    Macros = State#state.macros,
-    MacrosEnv = [{K, merl:term(V)} || K := V <- Macros],
     MacrosTree = merl:tsubst(AttrsTree, MacrosEnv),
-    AllVars = merl:template_vars(merl:template(MacrosTree)),
-    case AllVars -- maps:keys(Macros) of
-        [] when Eval ->
+    Vars = merl:template_vars(merl:template(MacrosTree)),
+    case Vars =:= [] andalso Eval of
```

Comment:
Did you try `case {Vars, Eval} of` and then a different set of case clauses, for readability?

Reply:
In that case, I need a boolean value, so I don't see a readability issue here.

Reply:
Check

```erlang
    case Vars =:= [] andalso Eval of
        true ->
            ... something;
        false ->
            ... something else
    end
```

vs.

```erlang
    case {Vars, Eval} of
        {[], true} ->
            ... something;
        _Other ->
            ... something else
    end
```

In the first bit of code I have to reason on the shortcut. In the second bit of code I know exactly (and in the same line) what `Vars` and `Eval` and both expected to be.

Better still would be to name that condition (`Vars =:= [] andalso Eval`) because with a bit less context you won't know what it stands for.

Reply:
Thanks for the suggestion; I agree. The second code is much easier for my eyes to follow visually, while the first code requires a little more attention to understand.

Reply:
[59d551a](https://github.com/arizona-framework/arizona/pull/177/commits/59d551adf070235c838d5651763efe90f5d2ae97)